agent/connect/ca/provider_vault.go

```diff
@@ -65,10 +74,71 @@ func (v *VaultProvider) Configure(cfg ProviderConfig) error {
 	v.isPrimary = cfg.IsPrimary
 	v.clusterID = cfg.ClusterID
 	v.spiffeID = connect.SpiffeIDSigningForCluster(&structs.CAConfiguration{ClusterID: v.clusterID})
+	v.shutdownCh = make(chan struct{}, 0)
+
+	// Look up the token to see if we can auto-renew its lease.
+	secret, err := client.Auth().Token().Lookup(config.Token)
+	if err != nil {
+		return err
+	}
+	var token struct {
+		Renewable bool
+		TTL       int
+	}
+	if err := mapstructure.Decode(secret.Data, &token); err != nil {
+		return err
+	}
+
+	// Set up a renewer to renew the token automatically, if supported.
+	if token.Renewable {
+		renewer, err := client.NewRenewer(&vaultapi.RenewerInput{
+			Secret: &vaultapi.Secret{
+				Auth: &vaultapi.SecretAuth{
+					ClientToken:   config.Token,
+					Renewable:     token.Renewable,
+					LeaseDuration: secret.LeaseDuration,
+				},
+			},
+			Increment: int(token.TTL),
+		})
+		if err != nil {
+			return fmt.Errorf("Error beginning Vault provider token renewal: %v", err)
+		}
+		go v.renewToken(renewer)
+	}
 
 	return nil
 }
 
+// renewToken uses a vaultapi.Renewer to repeatedly renew our token's lease.
+func (v *VaultProvider) renewToken(renewer *vaultapi.Renewer) {
+	go renewer.Renew()
+
+	for {
+		select {
+		case <-v.shutdownCh:
+			renewer.Stop()
+			return
+
+		case err := <-renewer.DoneCh():
+			if err != nil {
+				v.logger.Error(fmt.Sprintf("Error renewing token for Vault provider: %v", err))
```

Comment:
Reading https://godoc.org/github.com/hashicorp/vault/api#LifetimeWatcher, it sounds like in this case the renewer will no longer do anything. 

Should we attempt to restart the renewer here? If not we can probably exit the function since nothing else will happen.